internal/runbits/legacy/projectmigration/projectmigration.go

```diff
@@ -1,185 +1,84 @@
-// commitmediator.Get() for retrieving legacy commitId from activestate.yaml.
-// This is an anti-pattern and is only used to make this legacy feature palatable.
-func Register(prompter_ prompt.Prompter, out_ output.Outputer) {
-	prompter = prompter_
-	out = out_
+type migrator struct {
+	out  output.Outputer
+	proj projecter
 }
 
-func PromptAndMigrate(proj projecter) error {
-	if prompter == nil || out == nil {
-		return errs.New("projectmigration.Register() has not been called")
-	}
+func New(out output.Outputer, pj projecter) *migrator {
+	return &migrator{out: out, proj: pj}
+}
 
-	if os.Getenv(constants.DisableProjectMigrationPrompt) == "true" {
+// setupProject will ensure that the stored project matches the path that was requested. In most cases this should match,
+// and setting up a new project instance will be rare.
+func (m *migrator) setupProject(pjpath string) error {
+	if !ptr.IsNil(m.proj) && m.proj.Dir() == pjpath {
 		return nil
 	}
-
-	// We always set the local commit, the migration only touches on what happens with the commit in the activestate.yaml
-	if err := localcommit.Set(proj.Dir(), proj.LegacyCommitID()); err != nil {
-		return errs.Wrap(err, "Could not create local commit file")
-	}
-	for dir := proj.Dir(); filepath.Dir(dir) != dir; dir = filepath.Dir(dir) {
-		if !fileutils.DirExists(filepath.Join(dir, ".git")) {
-			continue
-		}
-		err := localcommit.AddToGitIgnore(dir)
-		if err != nil {
-			if !errors.Is(err, fs.ErrPermission) {
-				multilog.Error("Unable to add local commit file to .gitignore: %v", err)
-			}
-			out.Notice(locale.T("notice_commit_id_gitignore"))
-		}
-		break
-	}
-
-	// Prevent also showing the warning when we already prompt
-	warned = true
-
-	defaultChoice := false
-	if migrate, err := prompter.Confirm("", locale.T("projectmigration_confirm"), &defaultChoice); err == nil && !migrate {
-		if out.Config().Interactive {
-			out.Notice(locale.T("projectmigration_declined"))
-		}
-		return CreateMigrateScript(proj)
-	} else if err != nil {
-		return errs.Wrap(err, "Could not confirm migration choice")
-	}
-
-	if err := proj.StripLegacyCommitID(); err != nil {
-		return errs.Wrap(err, "Could not strip legacy commit ID")
-	}
-
-	out.Notice(locale.Tl("projectmigration_success", "Your project was successfully migrated"))
-
-	return nil
+	var err error
+	m.proj, err = project.FromPath(pjpath)
+	return err
 }
 
-var scriptsRx = regexp.MustCompile(`(?m)^scripts:\n`)
-
-func CreateMigrateScript(proj projecter) error {
-	scriptValue := migrateNixScript
-	scriptLanguage := "bash"
-	switch runtime.GOOS {
-	case "darwin":
-		scriptValue = migrateMacScript
-	case "windows":
-		scriptValue = migrateWinScript
-		scriptLanguage = "powershell"
+// Migrate returns the given project's commit ID in either the new format (commit file), or the old
+// format (activestate.yaml).
+// If you require the commit file, use localcommit.Get().
```

Comment:
This line is leftover from a previous comment and no longer applies. I would probably update the entire comment anyway, since as it is written, it implies you might call this function for the express purpose of obtaining a commit ID, rather than performing a migration step.